builtin/staker/sequence_test.go

```diff
@@ -0,0 +1,545 @@
+	"github.com/vechain/thor/v2/builtin/staker/delegation"
+	"github.com/vechain/thor/v2/builtin/staker/validation"
+	"github.com/vechain/thor/v2/thor"
+)
+
+type TestSequence struct {
+	staker *testStaker
+	t      *testing.T
+}
+
+func newTestSequence(t *testing.T, staker *testStaker) *TestSequence {
+	return &TestSequence{staker: staker, t: t}
+}
+
+func (ts *TestSequence) AssertActive(active bool) *TestSequence {
+	isActive, err := ts.staker.IsPoSActive()
+	assert.NoError(ts.t, err, "failed to check PoS active state")
+	assert.Equal(ts.t, active, isActive, "PoS active state mismatch")
+	return ts
+}
+
+func (ts *TestSequence) AssertLockedVET(expectedVET, expectedWeight uint64) *TestSequence {
+	locked, weight, err := ts.staker.LockedStake()
+	assert.NoError(ts.t, err, "failed to get locked VET")
+	assert.Equal(ts.t, expectedVET, locked, "locked VET mismatch")
+	assert.Equal(ts.t, expectedWeight, weight, "locked weight mismatch")
+
+	return ts
+}
+
+func (ts *TestSequence) AssertQueuedVET(expectedVET uint64) *TestSequence {
+	queued, err := ts.staker.QueuedStake()
+	assert.NoError(ts.t, err, "failed to get queued VET")
+	assert.Equal(ts.t, expectedVET, queued, "queued VET mismatch")
+
+	return ts
+}
+
+func (ts *TestSequence) AssertFirstActive(expectedAddr thor.Address) *TestSequence {
+	firstActive, err := ts.staker.FirstActive()
+	assert.NoError(ts.t, err, "failed to get first active validator")
+	assert.Equal(ts.t, expectedAddr, firstActive, "first active validator mismatch")
+	return ts
+}
+
+func (ts *TestSequence) AssertFirstQueued(expectedAddr thor.Address) *TestSequence {
+	firstQueued, err := ts.staker.FirstQueued()
+	assert.NoError(ts.t, err, "failed to get first queued validator")
+	assert.Equal(ts.t, expectedAddr, firstQueued, "first queued validator mismatch")
+	return ts
+}
+
+func (ts *TestSequence) AssertQueueSize(expectedSize uint64) *TestSequence {
+	size, err := ts.staker.QueuedGroupSize()
+	assert.NoError(ts.t, err, "failed to get queue size")
+	assert.Equal(ts.t, expectedSize, size, "queue size mismatch")
+	return ts
+}
+
+func (ts *TestSequence) AssertLeaderGroupSize(expectedSize uint64) *TestSequence {
+	size, err := ts.staker.LeaderGroupSize()
+	assert.NoError(ts.t, err, "failed to get leader group size")
+	assert.Equal(ts.t, expectedSize, size, "leader group size mismatch")
+	return ts
+}
+
+func (ts *TestSequence) AssertNext(prev thor.Address, expected thor.Address) *TestSequence {
+	next, err := ts.staker.Next(prev)
+	assert.NoError(ts.t, err, "failed to get next validator after %s", prev.String())
+	assert.Equal(ts.t, expected, next, "next validator mismatch after %s", prev.String())
+	return ts
+}
+
+func (ts *TestSequence) FirstActive() (thor.Address, *validation.Validation) {
+	first, err := ts.staker.FirstActive()
+	assert.NoError(ts.t, err)
+	val, err := ts.staker.GetValidation(first)
+	assert.NoError(ts.t, err)
+	return first, val
+}
+
+func (ts *TestSequence) AddValidation(
+	endorser, master thor.Address,
```

Comment:
Nit, might be worth using the same names as of the `staker.AddValidation` method signature.

endorsor -> validator
master -> endorsor